src/app/client/src/app/plugins/profile/components/profile-page/profile-page.component.ts

```diff
@@ -231,20 +231,24 @@ export class ProfilePageComponent implements OnInit, OnDestroy, AfterViewInit {
     });
   }
 
-  downloadCert(certificates) {
-    _.forEach(certificates, (value, key) => {
-      if (key === 0) {
-        if (_.get(value, 'identifier')) {
-          this.courseCService.getSignedCourseCertificate(_.get(value, 'identifier')).subscribe((resp) => {
-            this.certDownloadAsPdf.download(resp.printUri, null, _.get(value, 'name') );
-          }, error => {
-            this.downloadPdfCertificate(value);
-          });
-         } else {
-          this.downloadPdfCertificate(value);
-        }
+  downloadCert(course) {
+    // Check for V2
+    if (_.get(course, 'issuedCertificates.length')) {
+      const certificateInfo = course.issuedCertificates[0];
+      if (_.get(certificateInfo, 'identifier')) {
+        this.courseCService.getSignedCourseCertificate(_.get(certificateInfo, 'identifier')).subscribe((resp) => {
+          this.certDownloadAsPdf.download(resp.printUri, null, _.get(certificateInfo, 'name'));
+        }, error => {
+          this.downloadPdfCertificate(certificateInfo);
+        });
+      } else {
+        this.downloadPdfCertificate(certificateInfo);
       }
-    });
+    } else if (_.get(course, 'certificates.length')) { // For V1 - backward compatibility
```

Comment:
You have to check for length > 0.  this.downloadPdfCertificate method is not handled for undefined input value.